code/modules/reagents/reagent_containers/pill.dm

```diff
@@ -181,5 +181,17 @@
 	name = "speedy pill"
 	list_reagents = list("aranesp" = 10)
 
-
-
+/obj/item/reagent_containers/pill/floorpill
+	name = "floorpill"
+	desc = "A strange pill found in the depths of maintenance"
+	icon_state = "pill21"
+	var/static/list/names = list("maintenance pill","floorpill","mystery pill","suspicious pill","strange pill")
+	var/static/list/descs = list("Your feeling is telling you no, but...","Drugs are expensive, you can't afford not to eat any pills that you find."\
+	, "Surely, there's no way this could go bad.")
+
+/obj/item/reagent_containers/pill/floorpill/Initialize()
+	list_reagents = list(get_random_reagent_id() = rand(10,50))
+	name = pick(names)
+	if(prob(20))
+		desc = pick(descs)
+	. = ..()

```

Comment:
There's not reason you need to have your init code before the parent call here, except maybe the list_reagents? Also if you're going to have it on the last line you should be doing `return ..()`

Reply:
Needs to be there due to list_reagents, also I used . = ..() instead of return ..() because it looked cooler

Reply:
So just put it after list_reagents

Reply:
every time someone uses . = ..() instead of return ..() on the last line i cry a litttle